components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java

```diff
@@ -97,26 +97,15 @@ public class JDBCPermissionBasedInternalScopeValidator {
      */
     public String[] validateScope(OAuthTokenReqMessageContext tokReqMsgCtx) {
 
-        // filter internal scopes
-        String[] requestedScopes = Oauth2ScopeUtils.getRequestedScopes(tokReqMsgCtx.getScope());
+        String[] requestedScopes = tokReqMsgCtx.getScope();
```

Comment:
Here we aren't filtering for internal_* scope. Could this be an issue?

Reply:
As in previously any logic beyond this would only expect internal_* scopes. Now it might get some additional scopes.
Is this the case?

Reply:
No aiye. This method calls the getUserAllowedScopes() method. There we use Oauth2ScopeUtils.getRequestedScopes() to filter internal_* scopes.

Reply:
Refer to modified code from line 168 to 175.